### PR TITLE
Fixes #120 - Support for item horizontal margin zero with RangeItem

### LINKED ISSUE #120: Support for item horizontal margin zero with RangeItem
Using configuration option for item horizontal margin with zero doesn't respect the property.

Using option:
```javascript
margin: {item: {horizontal: 0, vertical: 10}}
```

The items should be displayed in one line, but instead:

image

Example code: https://jsfiddle.net/qu9t7dr2/

Tested with Chrome v77, with timeline v5.1 and the last version from repo

### PR DESCRIPTION
**Pull request by jasonex7**
Fixes #120

### CODE REVIEW

File: lib/timeline/component/item/RangeItem.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ class RangeItem extends Item {
  *                                  // TODO: describe options
  */
   constructor(data, conversion, options) {
-    super()
+    super();
```
Comment (mojoaxel) [inline lib/timeline/component/item/RangeItem.js]: It is not a goo style to change code that has nothing to do with this RP. Could you please remove this change from this RP. Maybe we will get a code-style fixer in the future that helps with this!

File: lib/timeline/component/item/RangeItem.js [hunk 2]
```diff
@@ -124,12 +124,12 @@ class RangeItem extends Item {
     this.overflow = window.getComputedStyle(this.dom.frame).overflow !== 'hidden';
     return {
       content: {
-        width: this.dom.content.offsetWidth,
+        width: this.dom.content.offsetWidth
```
Comment (mojoaxel) [inline lib/timeline/component/item/RangeItem.js]: see above

File: lib/timeline/component/item/RangeItem.js [hunk 3]
```diff
@@ -124,12 +124,12 @@ class RangeItem extends Item {
...
       },
       box: {
         height: this.dom.box.offsetHeight
       }
-    }
+    };
```
Comment (mojoaxel) [inline lib/timeline/component/item/RangeItem.js]: see above

File: lib/timeline/component/item/RangeItem.js [hunk 4]
```diff
@@ -246,8 +246,7 @@ class RangeItem extends Item {
       }
     }
 
-    // add 0.5 to compensate floating-point values rounding
-    const boxWidth = Math.max(end - start + 0.5, 1);
+    const boxWidth = Math.max(end - start, 1);
```
Comment (mojoaxel) [inline lib/timeline/component/item/RangeItem.js]: @yotamberk should have a look at this. He added the 0.5 compensation.

Comment (yotamberk) [reply lib/timeline/component/item/RangeItem.js]: I fixed this originally to compensate on half pixels rounding.
Why is this removed in this PR?

Comment (jasonex7) [reply lib/timeline/component/item/RangeItem.js]: It doesn't work with item horizontal margin 0. It shouldn't be necessary with `box-sizing: border-box;`

https://github.com/visjs/vis-timeline/blob/46ea275b053e2d6bbf3821460a8fec488aa1b47c/lib/timeline/component/css/item.css#L47-L50

Comment (yotamberk) [reply lib/timeline/component/item/RangeItem.js]: the CSS is not good enough. 
Checkout https://github.com/almende/vis/issues/2092
The correct solution would be to change the code to:
` const boxWidth = Math.max(Math.ceil(end - start), 1);`

Comment (jasonex7) [reply lib/timeline/component/item/RangeItem.js]: > The correct solution would be to change the code to:
> `const boxWidth = Math.max(Math.ceil(end - start), 1);`

That solution doesn't work with item horizontal margin zero. 

The case in almende/vis#2092 is a corner case when start and end dates of very few items differs only for milliseconds while majority differs in much more time. That few items fall in `Math.max` function to be less than 1 and then width of 1px is fixed without considerate start nor end dates and then the visualization are affected (blank spaces). To resolve this a propose rounding to 3 decimals instead.

**Pull request merged**